scipy/linalg/matfuncs.py

```diff
@@ -668,3 +672,191 @@ def rounded_sign(x):
+    Hazewinkel, Michiel, ed. (2001) [1994], "Symmetric matrix", Encyclopedia of Mathematics, 
+    Springer Science+Business Media B.V. / Kluwer Academic Publishers, ISBN 978-1-55608-010-4
+    credit to: Nils Werner
+    https://stackoverflow.com/questions/42908334/checking-if-a-matrix-is-symmetric-in-numpy
+    """
+    A = _asarray_square(A)
+    return np.allclose(A, A.T, rtol=rtol, atol=atol)
+
+def is_matrix_positive_definite(A):
+    """
+    Returns True if a matrix is positive definite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is positive definite.
+
+    Examples
+    --------
+    from scipy.linalg import is_matrix_positive_definite
+    m = np.array([[5,1],[1,3]])
+    print(is_matrix_positive_definite(m))
+    m = np.array([[1, -1, 0],[-1, 5, 0],[0, 0, 7]])
+    print(is_matrix_positive_definite(m))
+
+    References
+    ----------
+    Definiteness of a matrix, From Wikipedia, the free encyclopedia
+    https://en.wikipedia.org/wiki/Positive-definite_matrix#Eigenvalues
+    """
+    A = _asarray_square(A)
+    if not is_matrix_symmetric(A):
+        raise ValueError('expected symmetric matrix')
+    eiVal, eiVecR=eig(A)
+    return (eiVal>0).all()
+
+def is_matrix_positive_semidefinite(A):
+    """
+    Returns True if a matrix is positive semidefinite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is positive semidefinite.
+
+    Examples
+    --------
+    from scipy.linalg import is_matrix_positive_semidefinite
+    m = np.array([[1,-1],[-1,1]])
+    print(is_matrix_positive_semidefinite(m))
+
+    References
+    ----------
+    Definiteness of a matrix, From Wikipedia, the free encyclopedia
+    https://en.wikipedia.org/wiki/Positive-definite_matrix#Eigenvalues
+    """
+    A = _asarray_square(A)
+    if not is_matrix_symmetric(A):
+        raise ValueError('expected symmetric matrix')
+    eiVal, eiVecR=eig(A)
+    return (eiVal>=0).all()
+
+def is_matrix_negative_definite(A):
+    """
+    Returns True if a matrix is negative definite
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is negative definite.
+
+    Examples
+    --------
+    from scipy.linalg import is_matrix_negative_definite
+    m = np.array([[-5,-1],[-1,-3]])
+    print(is_matrix_negative_definite(m))
+
+    References
+    ----------
+    Definiteness of a matrix, From Wikipedia, the free encyclopedia
+    https://en.wikipedia.org/wiki/positive-definite_matrix#Eigenvalues
+    """
+    A = _asarray_square(A)
+    if not is_matrix_symmetric(A):
+        raise ValueError('expected symmetric matrix')
+    eiVal, eiVecR=eig(A)
+    return (eiVal<0).all()
+
+def is_matrix_negative_semidefinite(A):
```

Comment:
This will fail for negative definite matrices

Reply:
Not anymore, now there is a tolerance argument.